src/core/api/backbone_router_api.cpp

```diff
@@ -44,8 +44,6 @@ using namespace ot;
 
 otError otBackboneRouterGetPrimary(otInstance *aInstance, otBackboneRouterConfig *aConfig)
 {
-    OT_ASSERT(aConfig != nullptr);
```

Comment:
Why are we removing this assert, but keeping it in `otBackboneRouterGetConfig()` below?

Reply:
Good point! We should keep this. We can remove the ones for which we use `AsCoreType()` (since it will do the assert check for us). Will change it in next push.